bentoml/_internal/frameworks/onnx.py

```diff
@@ -0,0 +1,403 @@
+            Signatures of predict methods to be used. If not provided, the signatures default to
+            ``{"run": {"batchable": False}}``. Because we are using :obj:``onnxruntime.InferenceSession``,
+            the only allowed method name is "run"
+            See :obj:`~bentoml.types.ModelSignature` for more details.
+        labels (``dict[str, str]``, optional):
+            A default set of management labels to be associated with the model. An example is
+            ``{"training-set": "data-1"}``.
+        custom_objects (``dict[str, Any]``, optional):
+            Custom objects to be saved with the model. An example is
+            ``{"my-normalizer": normalizer}``.
+
+            Custom objects are currently serialized with cloudpickle, but this implementation is
+            subject to change.
+        metadata (``dict[str, Any]``, optional):
+            Metadata to be associated with the model. An example is ``{"bias": 4}``.
+
+            Metadata is intended for display in a model management UI and therefore must be a
+            default Python type, such as ``str`` or ``int``.
+
+        ...
+    Returns:
+        :obj:`~bentoml.Model`: A tag that can be used to access the saved model from the BentoML model
+        store.
+    Example:
+    .. code-block:: python
+        import bentoml
+
+        import torch
+        import torch.nn as nn
+
+        class ExtendedModel(nn.Module):
+            def __init__(self, D_in, H, D_out):
+                # In the constructor we instantiate two nn.Linear modules and assign them as
+                #  member variables.
+                super(ExtendedModel, self).__init__()
+                self.linear1 = nn.Linear(D_in, H)
+                self.linear2 = nn.Linear(H, D_out)
+
+            def forward(self, x, bias):
+                # In the forward function we accept a Tensor of input data and an optional bias
+                h_relu = self.linear1(x).clamp(min=0)
+                y_pred = self.linear2(h_relu)
+                return y_pred + bias
+
+
+        N, D_in, H, D_out = 64, 1000, 100, 1
+        x = torch.randn(N, D_in)
+        model = ExtendedModel(D_in, H, D_out)
+
+        input_names = ["x", "bias"]
+        output_names = ["output1"]
+
+        tmpdir = "/tmp/model"
+        model_path = os.path.join(tmpdir, "test_torch.onnx")
+        torch.onnx.export(
+            model,
+            (x, torch.Tensor([1.0])),
+            model_path,
+            input_names=input_names,
+            output_names=output_names,
+        )
+
+        tag = bentoml.onnx.save_model("onnx_model", model_path, signatures={"run": "batchable": True})
+
+    """
+    context = ModelContext(
+        framework_name="onnx",
+        framework_versions={
+            "onnx": get_pkg_version("onnx"),
+            _onnxruntime_pkg: _onnxruntime_version,
+        },
+    )
+
+    if signatures is None:
+        signatures = {
+            "run": {"batchable": False},
+        }
```

Comment:
Is `run()` the only allowed signature? With run, calling it from a runner will need the following syntax.
```
runner.run.run()
```

Reply:
`run` is the entrypoint of a `InferenceSession`.
To avoid name collision, we should make this an alias for signature
```python
runner.infer.run()
```
We can frozen the signature `infer` and forbid user to add `run` as a signature keys. If users provide a signature keys then we add this special signature to their signature dictionary.

With this we just need to mention how to use it in our docs with .
Thoughts @ssheng @larme ?

Reply:
@aarnphm  currently I just use `runner.run.run()` and explain it in doc. Do you want to add `runner.infer.run()` because `runner.run.run()` sounds a little strange?

Reply:
Does `run` not get set as the default, meaning users can just use `runner.run()`?

Reply:
@sauyon no because the original `runner.run` is overridden by runner method `run`

Reply:
I guess `runner.run.run()` is confusing, since the first `run` is actually the signature of InferenceSession, where as the second `run` is the Runner's run

Reply:
will keep it this way now and allow `runner.run()` in later stage